scripts/_util.py

```diff
@@ -27,3 +36,33 @@ def validate_bytes(
         raise click.BadParameter("not a hex string")
 
     return decode_hex(value)
+
+
+def account_from_keyfile(keyfile: Path, password: str) -> LocalAccount:
+    with open(keyfile, "rt") as fp:
+        privkey = Account.decrypt(json.load(fp), password)
+    return cast(LocalAccount, Account.from_key(privkey))
+
+
+def web3_for_rpc(rpc: str, account: LocalAccount) -> Web3:
+    web3 = Web3(HTTPProvider(rpc))
+
+    web3.eth.set_gas_price_strategy(rpc_gas_price_strategy)
+    web3.middleware_onion.inject(geth_poa_middleware, layer=0)
+    web3.middleware_onion.add(construct_sign_and_send_raw_middleware(account))
+    web3.eth.default_account = account.address
+
+    return web3
+
+
+def setup_web3(
+    rpc: str, keystore_file: Path, password: str, deployment_dir: Path
+) -> tuple[LocalAccount, Web3, dict[str, Contract]]:
+    account = account_from_keyfile(keystore_file, password)
+    web3 = web3_for_rpc(rpc, account)
+    deployment_info = beamer.contracts.load_deployment_info(deployment_dir)
+
+    chain_id = ChainId(web3.eth.chain_id)
+    contracts = beamer.contracts.make_contracts(web3, deployment_info[chain_id])
+
+    return account, web3, contracts
```

Comment:
Do we really need this function?
I feel it does too much: it loads the keyfile, constructs a `Web3` instance, queries the chain ID, loads the deployment info and then constructs the contracts. The name is also not reflecting the functionality of this function.

I think just moving `account_from_keyfile` and `web3_for_rpc` to a common place already improves things a lot, to the point that we really do not this "setup-everything" function. What do you think?

Reply:
Yes I thought about the same thing last night, and are about to change it back. It's not really useful

Reply:
I composed account creation and web3 creation into one, because whenever we create an account we almost always want to create web3 as well. Let me know if I even should split up further, I dont really mind

Reply:
Yeah, I think we should just go without `setup_web3` -- it's ok to have to write a few lines extra now and then.